block_my_feedback.php

```diff
@@ -183,49 +204,60 @@ public static function fetch_marking(stdClass $user): ?array {
     /**
      * Return mod data - due date & require marking.
      *
-     * TODO - turnitin, quiz.
-     *
      * @param cm_info $mod
      * @param stdClass $assess
      */
     public static function get_mod_data($mod, $assess): ?stdClass {
-        global $CFG;
-        // Mods have different fields for due date, and require marking.
-        switch ($mod->modname) {
-            case 'assign':
-
-                // Check mod due date is relevant.
-                $duedate = self::duedate_in_range($mod->customdata['duedate']);
-                if (!$duedate) {
-                    return null;
-                }
+        global $CFG, $DB;
 
-                // Add dates.
-                $assess->unixtimestamp = $duedate;
-                $assess->duedate = date('jS M', $duedate);
-
-                // Require marking.
-                require_once($CFG->dirroot.'/mod/assign/locallib.php');
-                $context = context_module::instance($mod->id);
-                $assignment = new assign($context, $mod, $mod->course);
-                $assess->requiremarking = $assignment->count_submissions_need_grading();
-                if (!$assess->requiremarking) {
-                    return null;
-                }
-                $assess->markingurl = new moodle_url('/mod/'. $mod->modname. '/view.php',
-                    ['id' => $assess->cmid, 'action' => 'grader']
-                );
+        // Turnitin features a due date per part and therefore needs to be treated separately here.
+        if ($mod->modname === 'turnitintooltwo') {
+            if (!isset($assess->partid)) { // If there is no Turnitin part ID there is no due date.
+                return null;
+            }
+            $record = $DB->get_record('turnitintooltwo_parts', ['id' => $assess->partid], 'dtdue');
+            $duedate = $record->dtdue;
+        } else {
+            $duedate = feedback_tracker::get_duedate($mod);
+        }
+        $duedate = self::duedate_in_range($duedate);
+        if (!$duedate) {
+            return null;
+        }
 
-                // Return template data.
-                return $assess;
+        // Add dates.
+        $assess->unixtimestamp = $duedate;
+        $assess->duedate = date('jS M', $duedate);
 
-            // TODO - quiz - 'timeclose' ?.
-            case 'quiz':
-                return null;
-            // TODO - turnitin.
-            default:
-                return null;
+        // Require marking.
+        require_once($CFG->dirroot.'/mod/assign/locallib.php');
```

Comment:
I don't think there's a better way of dealing with this at the moment but core seem to be in the process of removing these lib files and splitting them off so that will likely need a change in a 5.x version

Reply:
I agree. There is a similar issue in feedback_tracker (classes/local/helper.php) to remember as well.